controllers/osbuildconfig_controller.go

```diff
@@ -47,13 +53,54 @@ type OSBuildConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *OSBuildConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling", "edgeDevice", req)
+
+	osBuildConfig, err := r.OSBuildConfigRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
 
-	// TODO(user): your logic here
+	if osBuildConfig.DeletionTimestamp != nil {
+		return ctrl.Result{}, nil
```

Comment:
I think we should delete all OSBuild CRs linked to this OSBuildConfig when the latter is deleted

Reply:
Sorry, this was supposed to be put here:
AFAIK you need to set the owner for that to happen.
See example usage here: https://github.com/ctera/kubefiler-operator/blob/2a06c8bd0f0b9ba04dd783c8d6c4867762fc4f87/internal/resources/deployment.go#L88
See documentation here: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference

Reply:
The OSBuild controller will be notified on the deleting of the CR. It will then need to understand what to do